### PR TITLE
[MISC] Rename reference_begin_pos to .._position.

### PR DESCRIPTION
**Pull request by smehringer**
Follow up after #1862

### CODE REVIEW

File: include/seqan3/search/detail/policy_search_result_builder.hpp [hunk 1]
```diff
@@ -92,7 +92,7 @@ struct policy_search_result_builder
         // sort by reference id or by reference position if both have the same reference id.
         std::sort(results.begin(), results.end(), [] (auto const & r1, auto const & r2)
         {
-            return (r1.reference_id() == r2.reference_id()) ? (r1.reference_begin_pos() < r2.reference_begin_pos())
+            return (r1.reference_id() == r2.reference_id()) ? (r1.reference_begin_position() < r2.reference_begin_position())
                                                             : (r1.reference_id() < r2.reference_id());
```
Comment (eseiler) [inline include/seqan3/search/detail/policy_search_result_builder.hpp:96]:
```suggestion
            return (r1.reference_id() == r2.reference_id()) ? (r1.reference_begin_position() < 
                                                               r2.reference_begin_position())
                                                            : (r1.reference_id() < r2.reference_id());
```

File: include/seqan3/search/search_result.hpp [hunk 2]
```diff
@@ -56,18 +56,18 @@ namespace seqan3
...
  * * seqan3::search_result::reference_id()
- * * seqan3::search_result::reference_begin_pos()
+ * * seqan3::search_result::reference_begin_position()
  *
  * Note that the index cursor is not included in a hit by default. If you are trying to use the respective member
  * function, a static_assert will prevent you from doing so. You can configure the result of the search with the
  * \ref search_configuration_subsection_output "output configuration".
  */
-template <typename query_id_type, typename cursor_type, typename reference_id_type, typename reference_begin_pos_type>
+template <typename query_id_type, typename cursor_type, typename reference_id_type, typename reference_begin_position_type>
```
Comment (eseiler) [inline include/seqan3/search/search_result.hpp:65]:
```suggestion
template <typename query_id_type,
          typename cursor_type,
          typename reference_id_type,
          typename reference_begin_position_type>
```

File: include/seqan3/search/search_result.hpp [hunk 3]
```diff
@@ -56,18 +56,18 @@ namespace seqan3
...
 //!\cond
     requires (std::integral<query_id_type> || std::same_as<query_id_type, detail::empty_type>) &&
              (fm_index_cursor_specialisation<cursor_type> || std::same_as<cursor_type, detail::empty_type>) &&
              (std::integral<reference_id_type> || std::same_as<reference_id_type, detail::empty_type>) &&
-             (std::integral<reference_begin_pos_type> || std::same_as<reference_begin_pos_type, detail::empty_type>)
+             (std::integral<reference_begin_position_type> || std::same_as<reference_begin_position_type, detail::empty_type>)
```
Comment (eseiler) [inline include/seqan3/search/search_result.hpp:70]:
```suggestion
             (std::integral<reference_begin_position_type> || std::same_as<reference_begin_position_type,
                                                                           detail::empty_type>)
```

File: test/unit/search/search_test.cpp [hunk 4]
```diff
@@ -24,7 +24,7 @@ using seqan3::operator""_phred42;
 
 using namespace std::string_literals;
 
-auto position = seqan3::views::persist | std::views::transform([] (auto && res) { return res.reference_begin_pos(); });
+auto position = seqan3::views::persist | std::views::transform([] (auto && res) { return res.reference_begin_position(); });
```
Comment (eseiler) [inline test/unit/search/search_test.cpp:27]:
```suggestion
auto position = seqan3::views::persist |
                std::views::transform([] (auto && res) { return res.reference_begin_position(); });
```

File: test/unit/core/algorithm/pipeable_config_element_selector_test_template.hpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
+// -----------------------------------------------------------------------------------------------------
```
Comment (eseiler) [inline test/unit/core/algorithm/pipeable_config_element_selector_test_template.hpp:1]: The file seems unrelated?
Or did I miss something?

Comment (smehringer) [reply test/unit/core/algorithm/pipeable_config_element_selector_test_template.hpp:1]: ops!

File: include/seqan3/search/detail/search_configurator.hpp [hunk 6]
```diff
@@ -58,14 +58,14 @@ class search_configurator
         using index_cursor_t = std::conditional_t<traits_type::output_index_cursor, index_cursor_type, empty_type>;
         //!\brief The reference_id type of the search_result.
         using reference_id_t = std::conditional_t<traits_type::output_reference_id, index_size_type, empty_type>;
-        //!\brief The reference_begin_pos type of the search_result.
-        using reference_begin_pos_t = std::conditional_t<traits_type::output_reference_begin_pos,
+        //!\brief The reference_begin_position type of the search_result.
+        using reference_begin_position_t = std::conditional_t<traits_type::output_reference_begin_position,
                                                          index_size_type,
                                                          empty_type>;
```
Comment (eseiler) [inline include/seqan3/search/detail/search_configurator.hpp:65]:
```suggestion
        using reference_begin_position_t = std::conditional_t<traits_type::output_reference_begin_position,
                                                              index_size_type,
                                                              empty_type>;

```
